src/background/extension.ts

```diff
@@ -537,6 +539,28 @@ export class Extension implements ExtensionState {
         this.changeSettings({siteList: toggledList});
     }
 
+    //------------------------------------
+    //
+    //    Fetch colors from local CSS
+    //
+    //
+
+    private cssUpdate() {
+        const rootVariables = getComputedStyle(document.documentElement);
+        if (this.user.settings.theme.cssVariableBg && /^#([0-9a-fA-F]{3}){1,2}$/.test(rootVariables.getPropertyValue(this.user.settings.theme.cssVariableBg).trim())){
```

Comment:
Please use the `isValidHexColor` util helper to check if it's a valid hex color.